File: back-end/src/main/java/se/lnu/vaxjovolunteers/models/HeartbeatResponse.java
package se.lnu.vaxjovolunteers.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

@JsonSerialize
public record HeartbeatResponse(
        @JsonProperty("google_id")
        String googleID,
        @JsonProperty("moderator_status")
        boolean isModerator
) {
}

File: front-end/App/lib/model/heartbeatResponse.dart
class HeartbeatResponse {
  final String google_id;
  final bool is_moderator;

  HeartbeatResponse(this.google_id, this.is_moderator);

  factory HeartbeatResponse.fromJson(dynamic json) {
    try {
      final google_id = json['google_id'] as String;
      final is_moderator = json['moderator_status'] as bool;

      return HeartbeatResponse(
          google_id, is_moderator);
    } catch (e) {
      print(e);
      throw Exception(e);
    }
  }
}
